src/pegasus/request.lua

```diff
@@ -91,12 +91,21 @@ function Request:parseFirstLine()
   self.querystring = self:parseUrlEncoded(querystring)
 end
 
-function Request:parseUrlEncoded(value)
+function Request:parseUrlEncoded(data)
   local output = {}
 
-  if value then
-    for k, v in  string.gmatch(value, Request.PATTERN_QUERY_STRING) do
-        output[k] = v
+  if data then
```

Comment:
Thanks again, @Tieske; nice job! :+1:
Could you please add a few unit tests for this new logic?

Reply:
Done!